datadog_lambda/dsm.py

```diff
@@ -1,38 +1,61 @@
-from datadog_lambda import logger
+import logging
+import json
 from datadog_lambda.trigger import EventTypes
 
+logger = logging.getLogger(__name__)
 
-def set_dsm_context(event, event_source):
 
+def set_dsm_context(event, event_source):
     if event_source.equals(EventTypes.SQS):
         _dsm_set_sqs_context(event)
 
 
 def _dsm_set_sqs_context(event):
-    from datadog_lambda.wrapper import format_err_with_traceback
-    from ddtrace.internal.datastreams import data_streams_processor
-    from ddtrace.internal.datastreams.processor import DsmPathwayCodec
-    from ddtrace.internal.datastreams.botocore import (
-        get_datastreams_context,
-        calculate_sqs_payload_size,
-    )
+    from ddtrace.data_streams import set_consume_checkpoint
 
     records = event.get("Records")
     if records is None:
         return
-    processor = data_streams_processor()
 
     for record in records:
-        try:
-            queue_arn = record.get("eventSourceARN", "")
-
-            contextjson = get_datastreams_context(record)
-            payload_size = calculate_sqs_payload_size(record)
-
-            ctx = DsmPathwayCodec.decode(contextjson, processor)
-            ctx.set_checkpoint(
-                ["direction:in", f"topic:{queue_arn}", "type:sqs"],
-                payload_size=payload_size,
-            )
-        except Exception as e:
-            logger.error(format_err_with_traceback(e))
+        arn = record.get("eventSourceARN", "")
+        context_json = _get_dsm_context_from_lambda(record)
+        if not context_json:
+            logger.debug("DataStreams skipped lambda message: %r", record)
+            return None
+
+        carrier_get = _create_carrier_get(context_json)
+        set_consume_checkpoint("sqs", arn, carrier_get)
```

Comment:
(Let's make sure we update the set manual_checkpoint=False here once the other PR is merged & the tracer released.